net/Socket.cpp

```diff
@@ -1395,6 +1407,60 @@ std::ostream& StreamSocket::stream(std::ostream& os) const
     return os << "]";
 }
 
+bool StreamSocket::checkRemoval(std::chrono::steady_clock::time_point now)
+{
+    if (!isIPType()) // forced removal on IPv[46] network connections only
+    {
+        return false;
+    }
+    const auto durTotal =
+        std::chrono::duration_cast<std::chrono::milliseconds>(now - getCreationTime());
+    const auto durLast =
+        std::chrono::duration_cast<std::chrono::milliseconds>(now - getLastSeenTime());
+    const double bytesPerSecIn = durTotal.count() > 0 ? (double)bytesRcvd() / ((double)durTotal.count() / 1000.0) : 0.0;
+    const bool c1 = now < getCreationTime();
+    const bool c2 = _pollTimeout > std::chrono::microseconds::zero() &&
+                    durLast > _pollTimeout;
+    const bool c3 = _minBytesPerSec > std::numeric_limits<double>::epsilon() &&
+                    bytesPerSecIn > std::numeric_limits<double>::epsilon() &&
+                    bytesPerSecIn < _minBytesPerSec;
+    const bool c4 = SigUtil::getTerminationFlag();
```

Comment:
and that can be directly in the if I think =)

Reply:
I prefer having these criteria flags dedicated for logging, but renamed then.